packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -922,27 +947,29 @@ export class AngularCompilerPlugin {
     // Make a new program and load the Angular structure.
     await this._createOrUpdateProgram();
 
-    // Try to find lazy routes if we have an entry module.
-    // We need to run the `listLazyRoutes` the first time because it also navigates libraries
-    // and other things that we might miss using the (faster) findLazyRoutesInAst.
-    // Lazy routes modules will be read with compilerHost and added to the changed files.
-    let lazyRouteMap: LazyRouteMap = {};
-    if (!this._JitMode || this._firstRun) {
-      lazyRouteMap = this._listLazyRoutesFromProgram();
-    } else {
-      const changedTsFiles = this._getChangedTsFiles();
-      if (changedTsFiles.length > 0) {
-        lazyRouteMap = this._findLazyRoutesInAst(changedTsFiles);
+    if (this._discoverLazyRoutes) {
+      // Try to find lazy routes if we have an entry module.
+      // We need to run the `listLazyRoutes` the first time because it also navigates libraries
+      // and other things that we might miss using the (faster) findLazyRoutesInAst.
+      // Lazy routes modules will be read with compilerHost and added to the changed files.
+      let lazyRouteMap: LazyRouteMap = {};
+      if (!this._JitMode || this._firstRun) {
+        lazyRouteMap = this._listLazyRoutesFromProgram();
+      } else {
+        const changedTsFiles = this._getChangedTsFiles();
+        if (changedTsFiles.length > 0) {
+          lazyRouteMap = this._findLazyRoutesInAst(changedTsFiles);
+        }
       }
-    }
 
-    // Find lazy routes
-    lazyRouteMap = {
-      ...lazyRouteMap,
-      ...this._options.additionalLazyModules,
-    };
+      // Find lazy routes
+      lazyRouteMap = {
+        ...lazyRouteMap,
+        ...this._options.additionalLazyModules,
```

Comment:
Should it still support the additional lazy modules?

Reply:
I don't think so. We do currently support this for VE. But it hinges on `SystemJsNgModuleLoaderConfig`, which will be deprecated for Ivy (https://github.com/angular/angular/pull/28685#issuecomment-473387576). So I think that for Ivy projects this functionality shouldn't work at all.

This in turn means we should warn that it's not working. Added a warning for that.